commit 6b51d13af95fce2842b63a5e9650cda23295bc3a
Author: Markus Bauer <mb@karo-electronics.de>
Date:   Wed Apr 22 14:10:06 2020 +0000

    Also add videomodes PIXCLK POLARITY flags to drm mode and reverse

diff --git a/drivers/gpu/drm/drm_modes.c b/drivers/gpu/drm/drm_modes.c
index d4d6451..dc941cd 100644
--- a/drivers/gpu/drm/drm_modes.c
+++ b/drivers/gpu/drm/drm_modes.c
@@ -601,6 +601,10 @@ void drm_display_mode_from_videomode(const struct videomode *vm,
 	dmode->clock = vm->pixelclock / 1000;
 
 	dmode->flags = 0;
+	if (vm->flags & DISPLAY_FLAGS_PIXDATA_POSEDGE)
+		dmode->flags |= DRM_BUS_FLAG_PIXDATA_DRIVE_POSEDGE;
+	if (vm->flags & DISPLAY_FLAGS_PIXDATA_NEGEDGE)
+		dmode->flags |= DRM_BUS_FLAG_PIXDATA_DRIVE_NEGEDGE;
 	if (vm->flags & DISPLAY_FLAGS_HSYNC_HIGH)
 		dmode->flags |= DRM_MODE_FLAG_PHSYNC;
 	else if (vm->flags & DISPLAY_FLAGS_HSYNC_LOW)
@@ -642,6 +646,10 @@ void drm_display_mode_to_videomode(const struct drm_display_mode *dmode,
 	vm->pixelclock = dmode->clock * 1000;
 
 	vm->flags = 0;
+	if (dmode->flags & DRM_BUS_FLAG_PIXDATA_DRIVE_POSEDGE)
+		vm->flags |= DISPLAY_FLAGS_PIXDATA_POSEDGE;
+	if (dmode->flags & DRM_BUS_FLAG_PIXDATA_DRIVE_NEGEDGE)
+		vm->flags |= DISPLAY_FLAGS_PIXDATA_NEGEDGE;
 	if (dmode->flags & DRM_MODE_FLAG_PHSYNC)
 		vm->flags |= DISPLAY_FLAGS_HSYNC_HIGH;
 	else if (dmode->flags & DRM_MODE_FLAG_NHSYNC)
